cmd/go-clang-gen/main_array_test.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/go-clang/gen"
+	"github.com/go-clang/gen/test/expectations"
+)
+
+var testAPI *gen.API = &gen.API{
+	PrepareFunctionName:     prepareFunctionName,
+	PrepareFunction:         prepareFunction,
+	FilterFunction:          filterFunction,
+	FilterFunctionParameter: filterFunctionParameter,
+	FixedFunctionName:       fixedFunctionName,
+
+	PrepareStructMembers:     prepareStructMembers,
+	FilterStructMemberGetter: filterStructMemberGetter,
+}
+
+var testDataFolder string = "../../test/testdata/"
+
+func TestDriver(t *testing.T) {
```

Comment:
Afaik, this is testing almost the same behaviour as main(), to please call the test function "TestMain" for now.